eureka-core/src/main/java/com/netflix/eureka/cluster/DynamicGZIPContentEncodingFilter.java

```diff
@@ -27,37 +29,29 @@ public DynamicGZIPContentEncodingFilter(EurekaServerConfig config) {
     }
 
     @Override
-    public ClientResponse handle(ClientRequest request) throws ClientHandlerException {
+    public ClientResponse handle(ClientRequest request) {
+        // If 'Accept-Encoding' is not set, assume gzip as a default
         if (!request.getHeaders().containsKey(HttpHeaders.ACCEPT_ENCODING)) {
-            request.getHeaders().add(HttpHeaders.ACCEPT_ENCODING, "gzip");
+            request.getHeaders().add(HttpHeaders.ACCEPT_ENCODING, GZIP_ENCODING);
         }
 
         if (request.getEntity() != null) {
-            Object o = request.getHeaders().getFirst(HttpHeaders.CONTENT_ENCODING);
-            if (o != null && o.equals("gzip")) {
-                request.setAdapter(new Adapter(request.getAdapter()));
+            Object requestEncoding = request.getHeaders().getFirst(HttpHeaders.CONTENT_ENCODING);
+            if (GZIP_ENCODING.equals(requestEncoding)) {
+                request.setAdapter(new GzipAdapter(request.getAdapter()));
             } else if (isCompressionEnabled()) {
-                request.getHeaders().add(HttpHeaders.CONTENT_ENCODING, "gzip");
-                request.setAdapter(new Adapter(request.getAdapter()));
+                request.getHeaders().add(HttpHeaders.CONTENT_ENCODING, GZIP_ENCODING);
+                request.setAdapter(new GzipAdapter(request.getAdapter()));
             }
         }
 
         ClientResponse response = getNext().handle(request);
 
-        if (response.hasEntity() &&
-                response.getHeaders().containsKey(HttpHeaders.CONTENT_ENCODING)) {
-            String encodings = response.getHeaders().getFirst(HttpHeaders.CONTENT_ENCODING);
-
-            if (encodings.equals("gzip")) {
-                response.getHeaders().remove(HttpHeaders.CONTENT_ENCODING);
-                try {
-                    response.setEntityInputStream(new GZIPInputStream(response.getEntityInputStream()));
-                } catch (IOException ex) {
-                    throw new ClientHandlerException(ex);
-                }
-            }
+        String responseEncoding = response.getHeaders().getFirst(HttpHeaders.CONTENT_ENCODING);
```

Comment:
the gzip encoding could be specified as a later param, which would fail the getFirst here.